### PR TITLE
Improve Ballerina Filters

### PR DESCRIPTION
**Pull request by riyafa**
>Resolve https://github.com/ballerina-platform/ballerina-lang/issues/8590

>Also include fixes to WebSocket upgrade
>- WebSocket upgrade resource is changed to an Http resource and it follows same standards as an http resource.

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    }
+
+    /**
+     * Check if the resource is an http resource and apply filter.
+     *
+     * @param resourceNode The resource to apply filter on if it is http
+     * @param env          the symbol environment
+     */
+    void invokeFilters(BLangResource resourceNode, SymbolEnv env) {
+        if (resourceNode.requiredParams.size() == 2 && "http".equals(resourceNode.requiredParams.get(
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java]: http and Listener should move to constants.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        }
+    }
+
+    private void addFilterStatements(BLangResource resourceNode, SymbolEnv env) {
+        BLangVariable filterContextVar = addFilterContextCreation(resourceNode, env);
+        addAssignmentAndForEach(resourceNode, filterContextVar);
+    }
+
+    /**
+     * Adds code {@code http:FilterContext _$$_filterContext = new (serviceTypeDef, "serviceName", "resourceName");}.
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java]: Add proper doc comments.

Comment (riyafa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java]: Fixed

File: stdlib/ballerina-http/src/main/ballerina/http/auth/authz_filter.bal [hunk 3]
```diff
@@ -68,25 +68,35 @@ public type AuthzFilter object {
                 authorized = true;
             }
         }
-        return createAuthzResult(authorized);
+        return isAuthzSuccessfull(listener, authorized);
+    }
+
+    public function filterResponse(Response response, FilterContext context) returns boolean {
```
Comment (arukshani) [inline stdlib/ballerina-http/src/main/ballerina/http/auth/authz_filter.bal]: If we are not using these two input paras can we get rid of it?

Comment (riyafa) [reply stdlib/ballerina-http/src/main/ballerina/http/auth/authz_filter.bal]: It is the signature of a `http:Filter` object's `filterResponse` method and should be as it is for the match to succeed

**Pull request merged**